lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.AirLoopHVAC.rb

```diff
@@ -320,6 +320,116 @@ def air_loop_hvac_allowable_system_brake_horsepower(air_loop_hvac)
+      oa_flow_m3_per_s = get_airloop_hvac_design_oa_from_sql(air_loop_hvac)
+    else
+      OpenStudio.logFree(OpenStudio::Info, 'openstudio.standards.AirLoopHVAC', "For #{air_loop_hvac.name}, DCV not applicable because it has no OA intake.")
+      return false
+    end
+    oa_flow_cfm = OpenStudio.convert(oa_flow_m3_per_s, 'm^3/s', 'cfm').get
+
+    any_zones_req_dcv = false
+    air_loop_hvac.thermalZones.sort.each do |zone|
+      if user_model_zone_demand_control_ventilation_required?(zone)
+        any_zones_req_dcv = true
+        break
+      end
+    end
+
+    return true if any_zones_req_dcv && (has_economizer || (oa_flow_cfm > 3000))
+
+    return false
+  end
+
+  # Check if a zone in user model needs to have DCV per zone related requiremends in ASHRAE 90.1-2019 6.4.3.8
+  # @author Xuechen (Jerry) Lei, PNNL
+  # @param thermal_zone [OpenStudio::Model::ThermalZone] the thermal zone
+  # @return [Boolean] flag of whether thermal zone in user model is required to have DCV
+  def user_model_zone_demand_control_ventilation_required?(thermal_zone)
+    dcv_zone_user_exception = thermal_zone.additionalProperties.getFeatureAsBoolean('zone user specified DCV exception').get
+    return false if dcv_zone_user_exception
+
+    # check the following conditions at zone level
+    # zone > 500 sqft AND design occ > 25 ppl/ksqft
+
+    area_served_m2 = 0
+    num_people = 0
+    thermal_zone.spaces.each do |space|
+      area_served_m2 += space.floorArea
+      num_people += space.numberOfPeople
+    end
+    area_served_ft2 = OpenStudio.convert(area_served_m2, 'm^2', 'ft^2').get
+    occ_per_1000_ft2 = num_people / area_served_ft2 * 1000
+
+    return true if (area_served_ft2 > 500) && (occ_per_1000_ft2 > 25)
+
+    return false
+  end
+
+  # Check if the air loop in baseline model needs to have DCV
+  #
+  # @author Xuechen (Jerry) Lei, PNNL
+  # @param air_loop_hvac [OpenStudio::Model::AirLoopHVAC] air loop
+  # @return [Boolean] flag of whether the air loop in baseline is required to have DCV
+  def baseline_air_loop_hvac_demand_control_ventilation_required?(air_loop_hvac)
+    any_zone_req_dcv = false
+    air_loop_hvac.thermalZones.each do |zone|
+      if baseline_thermal_zone_demand_control_ventilation_required?(zone)
+        any_zone_req_dcv = true
+      end
+    end
+    return any_zone_req_dcv # baseline airloop needs dcv if any zone it serves needs dcv
+  end
+
+  # Check if the thermal zone in baseline model needs to have DCV
+  #
+  # @author Xuechen (Jerry) Lei, PNNL
+  # @param thermal_zone [OpenStudio::Model::ThermalZone] the thermal zone
+  # @return [Boolean] flag of whether thermal zone in baseline is required to have DCV
+  def baseline_thermal_zone_demand_control_ventilation_required?(thermal_zone)
+    # zone needs dcv if user model has dcv and baseline does not meet apxg exception
+    if thermal_zone.additionalProperties.hasFeature("apxg no need to have DCV")
```

Comment:
I believe this should be based on "zone dcv required by 901". Eventually we will have a Proposed model, and that model will follow 90.1 requirements, even if the user model does not have DCV. Same for second .HasFeature below.